src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +238,287 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
+                                ShortcircuitAnalysisType.ONE_BUS;
+                        if (analysisType == ShortcircuitAnalysisType.ALL_BUSES) {
+                            networkModificationTreeService.updateShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                            updateType = NotificationService.UPDATE_TYPE_SHORT_CIRCUIT_FAILED;
+                        } else {
+                            networkModificationTreeService.updateOneBusShortCircuitAnalysisResultUuid(receiverObj.getNodeUuid(), null);
+                            updateType = NotificationService.UPDATE_TYPE_ONE_BUS_SHORT_CIRCUIT_FAILED;
+                        }
+                    } break;
+                    case VOLTAGE_INITIALIZATION : {
+                        networkModificationTreeService.updateVoltageInitResultUuid(receiverObj.getNodeUuid(), resultUuid);
+                        updateType = NotificationService.UPDATE_TYPE_VOLTAGE_INIT_FAILED;
+                    } break;
+                    case DYNAMIC_SIMULATION : {
+                        networkModificationTreeService.updateDynamicSimulationResultUuid(receiverObj.getNodeUuid(), null);
+                        updateType = NotificationService.UPDATE_TYPE_DYNAMIC_SIMULATION_FAILED;
+                    } break;
                 }
+
+                UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+                // send notification for failed computation
+                notificationService.emitStudyError(
+                        studyUuid,
+                        receiverObj.getNodeUuid(),
+                        updateType,
+                        errorMessage,
+                        userId);
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.toString());
             }
-        };
+        }
     }
 
-    @Bean
-    public Consumer<Message<String>> consumeSensitivityAnalysisStopped() {
-        return message -> {
-            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
-            if (receiver != null) {
-                NodeReceiver receiverObj;
-                try {
-                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
-
-                    LOGGER.info("Sensitivity analysis stopped for node '{}'", receiverObj.getNodeUuid());
+    public void consumeCalculationStopped(Message<String> msg, ComputationType computationType) {
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        if (receiver != null) {
+            NodeReceiver receiverObj;
+            try {
+                receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                LOGGER.info("{} stopped for node '{}'", computationType.getLabel(), receiverObj.getNodeUuid());
```

Comment:
may be this comment should be after the switch case ? or change it to '{} is stopping ...'  because the service is not stopped yet

Reply:
it's a channel consumption in study-server of the stopped message from the rabbitMQ queue then the calculation is already stopped in the dedicated calculation server.
But maybe it's a good idea to move this log or add one at the end of this try scope to trace that the resultUuid for this type of calculation is reset.

Reply:
Agreed, moved.